src/pages-and-resources/discussions/app-config-form/apps/shared/discussion-topics/DiscussionTopics.jsx

```diff
@@ -1,24 +1,28 @@
-import React, { useEffect } from 'react';
+import React, { useEffect, useContext } from 'react';
 import { useDispatch } from 'react-redux';
 import { Add } from '@edx/paragon/icons';
 import { Button } from '@edx/paragon';
 import { injectIntl, intlShape } from '@edx/frontend-platform/i18n';
 import { FieldArray, useFormikContext } from 'formik';
 import { v4 as uuid } from 'uuid';
-import PropTypes from 'prop-types';
 
 import messages from '../messages';
 import TopicItem from './TopicItem';
 import { updateValidationStatus } from '../../../../data/slice';
+import { LegacyConfigFormContext } from '../../legacy/LegacyConfigFormProvider';
 
-const DiscussionTopics = ({ intl, discussionTopicErrors }) => {
+const DiscussionTopics = ({ intl }) => {
   const {
     values: appConfig,
-    setFieldValue,
     validateForm,
   } = useFormikContext();
-  const { discussionTopics, divideDiscussionIds } = appConfig;
+  const { discussionTopics } = appConfig;
   const dispatch = useDispatch();
+  const {
+    discussionTopicErrors,
+    validDiscussionTopics,
+    setValidDiscussionTopics,
+  } = useContext(LegacyConfigFormContext);
 
   const isFormInvalid = discussionTopicErrors.some((error) => error === true);
```

Comment:
is there any reason we have not shifted it to the parent component? I remember we discussed shifting it to the parent so that in the future we need to add validation on more fields.

Reply:
we decided to move discussionTopicErrors into parents as global error state. isFormInvalid logic only based on discussionTopicErrors global state. I do not think so we need this to move on parent.

Reply:
yes this is specific to Discussion Topics, We can move this to parent when we need to take other fields into account e.g. discussion blackout dates.